crates/node-db/src/lib.rs

```diff
@@ -316,7 +348,7 @@ pub fn get_block(
         }
         let mut block = block_in_progress.expect("should have been set above at worst case");
         // Add the solution.
-        let solution: Solution = decode(&solution_blob)?;
+        let solution = get_solution(tx, &solution_addr)?;
```

Comment:
Maybe add a comment that says something like: `get_solution` internally prepares statements for various tables. If we notice performance issue with `get_block` we should instead pass in and reuse the prepared statements.